_posts/2021-04-30-soliditylangorg-umbrella-domain.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+published: true
+title: 'What happened with Solidity-related Domains?'
+date: '2021-04-30'
+author: Franziska Heintel
+category: Explainers
+---
+
+Some time ago we decided to get a domain that the Solidity team has easy access to in
+order to streamline efforts and initiatives that were hosted on other domains before.
+
+And so, ``soliditylang.org`` was born! 🎉
+
+We announced most of these domain changes individually [on Twitter](https://twitter.com/solidity_lang/status/1329031139495964673?s=20), but we want to take a moment to
+also officially announce it here on the blog
+and explain the various subdomains we have now, what they are for and which older
+domains may be deprecated in future.
+
+In short, we currently maintain the following pages under the soliditylang.org umbrella:
+
+- Solidity Portal: [soliditylang.org](https://soliditylang.org/)
+- Documentation: [docs.soliditylang.org](https://docs.soliditylang.org/)
+- Blog: [blog.soliditylang.org](https://blog.soliditylang.org/)
+- Static Binaries: [binaries.soliditylang.org](https://binaries.soliditylang.org/)
+- Forum: [forum.soliditylang.org](https://forum.soliditylang.org/)
+- Solidity Summit: [summit.soliditylang.org](https://summit.soliditylang.org/)
+- Solidity Underhanded Contest: [underhanded.soliditylang.org](https://underhanded.soliditylang.org/)
+
+## Solidity Portal
+
+The Solidity language portal at [soliditylang.org](https://soliditylang.org/) is a one-stop-shop for all important links and high-level information around Solidity. It includes information on the latest Solidity release, how to get started, information on language design and features recent blog posts and news.
+## Documentation
+
+The Solidity documentation was originally available at ``solidity.readthedocs.io``. Now, the readthedocs domain redirects to the documentations' new home: [docs.soliditylang.org](https://docs.soliditylang.org/). Since it's a 301 redirect, all old links are still working and will redirect directly to the correct section in the docs on the new domain.
+## Blog
+
+The Solidity blog was originally available at ``solidity.ethereum.org``. Now, the blog is at home at [blog.soliditylang.org](https://blog.soliditylang.org/) and similarly to how it is  handled with the docs, the transition should have felt seamless to users. All old links still work and redirect to the corresponding equivalent on the new domain.
+## Static Binaries / Solc-Bin
```

Comment:
Yeah, if we want to describe it all in detail. But do we?

Reply:
I think it is an important thing worth its own post. Even if it is three paragraphs only it is worth it given compiler binaries were replaced. That is serious.

But even describing the asm.js and wasm binaries (what are the differences, when we started with which, etc.) can be quite some detail.

Reply:
Ok. I'm fine with it either way.

Reply:
> They're faster and for that reason @ekpyron rebuilt older binaries in solc-bin to wasm.

The reason was rather that browsers decreased the available javascript stack size, which caused the old builds to break by the way. *If* the wasm builds are actually noticable faster, then that's secondary to actually working :-).

Reply:
Ok maybe this topic should get its own short post then?

Reply:
> The reason was rather that browsers decreased the available javascript stack size, which caused the old builds to break by the way.

Interesting detail. That would explain why Remix is properly using wasm builds when available but some other projects that I've seen don't care and just to straight to `bin/`.

Reply:
In case you're interested: before moving to wasm builds we had interesting workarounds like this: https://github.com/ethereum/solidity/pull/6595 and that https://github.com/ethereum/solidity/pull/6595/files, since we figured out the part triggering the stack issues. Interestingly it wasn't only once but several times that browsers decreased those limits :-). But we didn't want to just backport those fixes and we weren't sure if and when it'd break again - moving to wasm and rebuilding all old versions was mainly a solution to that entire mess.